andela/travel_tool_front
# #161697793 Fix unmarked checkbox on editing a travel checklist item

#### What does this PR do?
- fix the state of the checkbox to be dependent on the previous operation carried on it.

#### Description of Task to be completed?
- The state of the checkbox should be dependent on the previous operation carried on it i.e. if it was checked or left unchecked

#### How should this be manually tested?
- Login to the application
- Navigate to the trip planner page as a travel admin
- Add a travel checklist item
- Make sure to mark the checkbox having a label of Require file attachment and then, save the item
- Try to edit item
- The checkbox is now dependent on the previous operation carried i.e. if it was checked, it would remain checked and left unchecked if it was not checked.

#### Any background context you want to provide?
- `requiresFiles` value now passed as a props to the CheckBox and not through the state via `values`

#### What are the relevant pivotal tracker stories?

[#161697793](https://www.pivotaltracker.com/story/show/161651815)
#### Screenshots (if appropriate)
